### PR TITLE
refactor: deprecate JSON interface from SDK

### PR DESCRIPTION
**Pull request by jjleng**
### Description
We should deprecate the JSON interface for transaction submission. Submitting transactions in JSON would require the server to generate signing messages, which is not safe. Transactions submissions should be done through the BCS interface. 

This PR added `@deprecated` tag to the JSON interfaces. Developers have visual indications in IDE. We are going to remove these interfaces in next major release.

### Test Plan
All the tests still pass.

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2374)

### CODE REVIEW

File: ecosystem/typescript/sdk/src/aptos_client.ts [hunk 1]
```diff
@@ -355,6 +360,7 @@ export class AptosClient {
   }
 
   /**
+   * @deprecated
```
Comment (gregnazario) [inline ecosystem/typescript/sdk/src/aptos_client.ts:363]: Do you have to say what it's been replaced with?  / should we do that?

**Review by gregnazario — APPROVED** [review]

**Comment by CapCap** [comment]
Deprecating this makes me a bit sad... The existing BCS interface is still pretty raw and rough; is there a world where instead we add an endpoint to the node that returns a json representation of the method ABI instead, and locally convert to BCS based on that + sign?

**Comment by jjleng** [comment]
@CapCap thanks for sharing your concerns. This PR won't get landed. But it is still worth explaining what we are doing on the SDK side.
This PR only deprecates that JSON txn submission. For txn submission, we now have ABI support. Although we have endpoints to return the ABI in JSON format for methods, we followed the Eth web3 approach. We ask users to provide ABI on the client side. Users get ABIs through compiling Move.

For the read path, we are not deprecating the JSON payloads. The BCS payload reading and decoding are still a work in progress.

**Comment by jjleng** [comment]
This is not relevant any more because of the massive refactoring

**Pull request closed without merging**